experimental/internal/taxa/set.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package taxa
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bufbuild/protocompile/internal/iter"
+	"github.com/bufbuild/protocompile/internal/iters"
+)
+
+// Set is a set of [Subject] values, implicitly ordered by the [Subject] values'
+// intrinsic order.
+//
+// A zero Set is empty and ready to use.
+type Set struct {
+	bits [(Total + 63) / 64]uint64
+}
+
+// NewSet returns a new [Set] with the given values set.
+//
+// Panics if any value is not one of the constants in this package.
+func NewSet(whats ...Subject) Set {
```

Comment:
How about `items`, `elements`, or `subjects`? Just not `whats` 😝

Reply:
Well, the type was originally `What`. But I think subjects is best here.